src/Routing/Duplex.purs

```diff
@@ -79,12 +89,41 @@ suffix (RouteDuplex enc dec) s = RouteDuplex (\a -> enc a <> Printer.put s) (dec
 path :: forall a b. String -> RouteDuplex a b -> RouteDuplex a b
 path = flip (foldr prefix) <<< String.split (Pattern "/")
 
+-- | Modifies given `codec` by requiring it to be prefixed with '/'.
+-- | You can think of it as stripping the '/' at the beginning of route, failing if it's not there.
+-- |
+-- |```purescript
+-- | parse (root segment) "/abc" == Right "abc"
+-- | parse (root segment) "abc" == Left (Expected "" "abc")
+-- | print (root segment) "abc" == "/abc"
+-- |```
 root :: forall a b. RouteDuplex a b -> RouteDuplex a b
 root = path ""
 
+-- | Modifies given `codec` by requiring it to consume the entire rest of input.
+-- | That is `end codec` succeeds if `codec` suceeds AND all the
+-- | remaining segments have been consumed in the process.
+-- |
+-- |```purescript
+-- | parse (end segment) "abc" == Right "abc"
+-- |
+-- | parse (end segment) "abc/def" == Left (ExpectedEndOfPath "def")
+-- |```
 end :: forall a b. RouteDuplex a b -> RouteDuplex a b
 end (RouteDuplex enc dec) = RouteDuplex enc (dec <* Parser.end)
 
+-- | Process path segment as a String.
```

Comment:
I don't think "process" is the right word.

> Consumes and prints a single path segment.

What do you think?

Reply:
It's kind of hard because of RouteDuplex being double purpose thing. `Consumes and prints`  sounds more specific than just process, but since you can never use a codec for both purposes at a given call site, I'd replace `and` with `or`.

Reply:
I went for "Consumes or prints a single path segment." in the end.